openbb_platform/providers/bls/openbb_bls/models/series.py

```diff
@@ -0,0 +1,218 @@
+        )
+        now = datetime.now()
+        start_year = (
+            query.start_date.year
+            if query.start_date
+            else (now - timedelta(weeks=52 * 3)).year
+        )
+        end_year = query.end_date.year if query.end_date else now.year
+        results: Dict = {"data": [], "messages": [], "metadata": {}}
+        messages: List = []
+
+        # The max number of symbols per request is 50.
+        # The max year range is 20.
+        # We chunk the request to handle the provided start/end date by the user.
+        def chunk_list(lst, chunk_size):
+            """Yield successive chunks from lst of size chunk_size."""
+            for i in range(0, len(lst), chunk_size):
+                yield lst[i : i + chunk_size]
+
+        def chunk_years(start_year, end_year, chunk_size):
+            """Yield successive year ranges of size chunk_size."""
+            for year in range(start_year, end_year + 1, chunk_size):
+                yield (year, min(year + chunk_size - 1, end_year))
+
+        # Define a function to wrap as a coroutine.
+        async def make_query(symbol, start, end):
+            """Make a query to the BLS API."""
+            data = await get_bls_timeseries(
+                api_key=api_key,
+                series_ids=symbol,
+                start_year=start,
+                end_year=end,
+                calculations=query.calculations,
+                catalog=True,
+                annual_average=query.annual_average,
+                aspects=query.aspects,
+            )
+            if isinstance(data, dict):
+                results.update(
+                    {
+                        "data": results.get("data", []) + data.get("data", []),
+                        "messages": list(
+                            set(results.get("messages", []) + data.get("messages", []))
+                        ),
+                        "metadata": {
+                            **results.get("metadata", {}),
+                            **data.get("metadata", {}),
+                        },
+                    }
+                )
+            elif isinstance(data, EmptyDataError) and data.message:
+                messages.append(data.__dict__.get("message", ""))
+
+        # Create a list of tasks to run based on the API query limitations.
+        tasks: List = []
+
+        for symbol_chunk in chunk_list(symbols, 50):
+            for year_range in chunk_years(start_year, end_year, 20):
+                tasks.append(
+                    asyncio.create_task(
+                        make_query(
+                            symbol_chunk,
+                            year_range[0],
+                            year_range[1],
+                        )
+                    )
+                )
+
+        await asyncio.gather(*tasks)
+
+        if not results.get("data"):
+            if messages:
+                _messages = ",".join(set(messages))
+            raise OpenBBError(_messages)
```

Comment:
Not an issue, just styling note that there is no need to define a new intermediate variable.

This also will throw an error if messages is empty because you define _messages within a single if condition